src/runner.ts

```diff
@@ -0,0 +1,578 @@
+  // refs in the  repo expression if present.
+  latestRelease?: boolean;
+
+  // An array of repo expressions defining the set of repos to require/load
+  // but not specifically to test.
+  repos?: string[];
+
+  // An array of repo expressions representing repos to exclude from testing
+  // should any matching ones be encountered in the tests array.
+  skipTests?: string[];
+
+  // An array of repo expressions defining the set of repos to test with the
+  // web-component-tester.  Note that repos in this list do not have to be
+  // present in the repos array.
+  tests: string[];
+
+  // If true, output will information used primarily for debug purposes.
+  verbose?: boolean;
+
+  // Command-line flags to send to web-component-tester.
+  wctFlags?: string[];
+
+  // The folder to clone repositories into and run tests from.  Defaults to
+  // './repos' if not provided.
+  workspaceDir?: string;
+}
+
+export class Runner {
+  // The repository patterns we do not want to load.
+  private _excludeRepos: string[];
+
+  // Always clone a fresh copy of the repository (don't just update existing
+  // clone.)
+  private _fresh: boolean;
+
+  private _github?: git.GitHubConnection;
+  private _repos: string[];
+  private _skipTests: string[];
+  private _tests: string[];
+  private _testRateLimiter: Bottleneck;
+  private _verbose: boolean;
+  private _wctFlags: string;
+  private _workspace: Workspace;
+
+  // TODO(usergenic): This constructor is getting long.  Break up some of
+  // these stanzas into supporting methods.
+  constructor(options: RunnerOptions) {
+    this._excludeRepos = options.excludeRepos || [];
+    this._fresh = !!options.fresh;
+    // TODO(usergenic): Pass an option to gitUtil.connectToGitHub for the
+    // rate limiter it uses.
+    this._github = new git.GitHubConnection(options.githubToken);
+    this._repos = options.repos || [];
+    this._skipTests = options.skipTests || [];
+    this._tests = options.tests || [];
+    this._verbose = !!options.verbose;
+    this._wctFlags = options.wctFlags ? options.wctFlags.join(' ') : '';
+    this._workspace = {
+      dir: (options.workspaceDir || './tattoo_workspace'),
+      repos: new Map()
+    };
+
+    // TODO(usergenic): Rate limit should be an option.
+    this._testRateLimiter = new Bottleneck(1, 100);
+
+    if (this._verbose) {
+      console.log('Tattoo Runner configuration:');
+      console.log({
+        excludeRepos: this._excludeRepos,
+        fresh: this._fresh,
+        repos: this._repos,
+        skipTests: this._skipTests,
+        tests: this._tests,
+        wctFlags: this._wctFlags,
+        workspaceDir: this._workspace.dir
+      });
+    }
+  }
+
+  /**
+ * Analyzes all of the HTML in 'repos/*' with hydrolysis.
+ *
+ * @returns a promise of the hydrolysis.Analyzer with all of the info loaded.
+ */
+  async _analyzeRepos() {
+    const dirs = fs.readdirSync(this._workspace.dir);
+    const htmlFiles: string[] = [];
+
+    for (const dir of dirs) {
```

Comment:
Moot.  Function removed in 9a88a6e0804a7b3da121d9f8944060c92af48f80 after it was discovered to be not being called anywhere.